### PR TITLE
Mech Balance V2

### PR DESCRIPTION
**Pull request by ghost**
:cl:
Balance:
-Mechs no longer explode on death
-Mechs and syndicate combat mechs sleep the pilot for 2s when destroyed, other combat mechs for 4s.
-Mechs now have exit times. 1s for basic ripley, 2s for other mechs, 4s for all combat mechs. You cannot punch or use equipment when exiting, and moving cancels.
/:cl:
Sleep on destruction and exit time are controlled by a variable of the mech.

### CODE REVIEW

**Comment by optimumtact** [comment]
what is the goal?

**Comment by py01** [comment]
currently mechs explode on destruction 30% of the time for (0,0,1,3) on the pilot. And, even if they were going to explode, you can instantly eject from the mech if it is going to go down in order to avoid the explosion. This allows you to pop out and immediately fight on foot with no consequences if your mech goes down. Given how high health and damage mechs are, I believe that if someone manages to turn the tables against a mech, they should have an opportunity to counterattack or arrest the pilot.

**Comment by zxaber** [comment]
More of a fluff thing, but a hissing hydraulic sound when exiting the mech would be neat. Some sort of tell that the pilot is bailing.

**Comment by py01** [comment]
@zxaber I'll put that in a seperate PR if I get around to making it.

**Comment by py01** [comment]
Actually, sound/machines/clockcult/steam_whoosh.ogg might be a good fit with a little editing

**Comment by 4dplanner** [comment]
Probably worth putting that the explosion chance is gone in the changelog

**Comment by py01** [comment]
@4dplanner fixed

**Comment by wesoda25** [comment]
What we really need is to have all mechs show up on the borg console

**Comment by Bawhoppen** [comment]
@wesoda25 Mech beacons can be made at robotics. I guess it's a player issue if the RD is authorizing mechs to be made without them...

**Comment by 4dplanner** [comment]
^this but unironically

**Comment by wesoda25** [comment]
So we could make beacons a necessary component in constructing all mechs.

**Comment by zxaber** [comment]
Balancing around mandatory remote kill switches is inherently unfun design.

**Comment by wesoda25** [comment]
Its also unfair design how through minimal effort one can become immune all crewmembers without armory access. The only mech which requires minimal risk, sacrifice, and luck is the phazon, which is so absurdly overpowered that it should require much greater sacrifice. 

Each mech should at least be able to be remotely locked down, they can always just deconstruct the console before hand (unless AI can do it if in camera view, not sure). I’m biased because I think mechs are complete shit game mechanics and code additions, but still think it would improve the game. 

Its not a matter of one person not being able to take a mech down, thats perfectly fine. Its the fact that a mob is helpless before mechs with any amount of upgrades, its simply unfair. You can call botany overpowered because they can make deathnettles n shit, security overpowered because they get weapons, xenobio overpowered because they are overpowered, but they are all succeptable to stuns, they aren’t literal Gods with armor like 3 r walls.

**Comment by PKPenguin321** [comment]
It would be pretty kick ass if mechs could be dismembered

**Comment by KomradeSpectre** [comment]
> 
> 
> It would be pretty kick ass if mechs could be dismembered

Kevinz has promised a mech overhaul with dismemberment as part of it, apparently after the Multi-Z stuff is done.

**Comment by GuyonBroadway** [comment]
To be even fairer when was the last time you saw a nukie use the marauder to any tangible effect? 

As someone who had had the chance to go on a mech joyride there are a few achillies heels that can put an end to your battletech beatdown.

Firstly flashbangs affect the mech pilot

**Comment by 4dplanner** [comment]
"mechs in-built ejector seat catapults you into the ceiling and have it accompanied by a comical sound effect"

no

on topic, from localhost testing the sleep duration feels a little small even for combat mechs, though obviously I'd have to see how it worked out in-game.

**Comment by py01** [comment]
I made the sleep just long enough for you to baton them, but it can be increased pretty easily after in-game testing if deemed to short.

**Comment by Dorsisdwarf** [comment]
so did the pr that removes mechs being instantly gibbed by ied lances go through or what

**Comment by zxaber** [comment]
IIRC, exploding mechs don't leave wreckage, so making them always explode removed a pretty nifty salvage thing (plus being able to remove AIs from dead mechs for restoring).

It would also mean no real way to non-lethally fight a mech, since every time the mech dies it would kill the pilot.

Anyway, the prior version of this PR had it explode every time, but I think this version is much better for everyone.

**Comment by wesoda25** [comment]
Yeah this is nice because it means borgs can actually fight mechs now, as opposed to trying to keep everyone away from them.

**Comment by necromanceranne** [comment]
Borgs are mechbane. They're like gnats and you can't stun the fuckers at all and they move quickly enough to get behind the mech and hit it's backside. My only concern is critical affects can still cause harm, like cabin fires, so it's not a total go-ahead for borgs to fight mechs. It can still potentially cause harm.

**Comment by zxaber** [comment]
Really depends on how it's taken. Borgs are allowed to harm Holoparasites, even though damage is passed back onto the (potentially) human host, because they're allowed to not know that. Similarly, they might be allowed to not know that cabin fires are a thing.

I still would like to see critical effects require a lower health than they do currently, because as it, there's like a 50% chance that half your mech's health bar is worthless.

**Comment by GuyonBroadway** [comment]
Doesn't this mean the thing to do before going on a mech joyride is just to chug as much coffee as you possibly can?

**Comment by ExcessiveUseOfCobblestone** [comment]
referencing ceilings is too dangerous

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -48,6 +48,9 @@
 	var/lights = FALSE
 	var/lights_power = 6
 	var/last_user_hud = 1 // used to show/hide the mecha hud while preserving previous preference
+	var/destruction_sleep_duration = 20 //sleep duration applied to pilot if mech is destroyed
+	var/is_currently_ejecting = FALSE //True when pilot is attempting to exit mech, disables equipment and punching when true
+	var/exit_delay = 20 //How long it takes pilot to eject from mech
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/mecha/mecha.dm]: these should be with enter delay or vice versa

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -161,6 +164,8 @@
 	return cell
 
 /obj/mecha/Destroy()
+	if(occupant!=null)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/mecha/mecha.dm]:
```suggestion
	if(occupant)
```

Comment (py01) [reply code/game/mecha/mecha.dm]: oh god web edit broke everything

**Comment by Borisvanmemes** [comment]
@KomradeSpectre I'm days late but iirc kev said that it won't happen due to him lacking sprites to do it, and it'd need FUCKLOADS of sprites.

**Comment by py01** [comment]
Is there a solution for an entire file becoming a diff when fixing merge conflicts? Not sure why it is occurring, since the two files have a bunch of identical lines.

**Comment by ExcessiveUseOfCobblestone** [comment]
line endings

**Comment by py01** [comment]
@ExcessiveUseOfCobblestone 
Alright this PR should be good to go now, I implemented your requested changes.

**Pull request merged**